fiftyone/core/dataset.py

```
@@ -1210,26 +1210,43 @@
         """Returns the first sample in the dataset.
 
         Returns:
             a :class:`fiftyone.core.sample.Sample`
         """
-        return super().first()
+        cursor = self._get_first()
+
+        try:
+            d = next(cursor)
+        except StopIteration:
+            raise ValueError("%s is empty" % self.__class__.__name__)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)